test/support/utils.rb

```diff
@@ -54,37 +52,41 @@ def create_pseudo_terminal(boot_options: "-r debug/run")
       timeout_sec = (ENV['RUBY_DEBUG_TIMEOUT_SEC'] || 10).to_i
 
       PTY.spawn("#{RUBY} #{boot_options} #{temp_file_path}") do |read, write, pid|
-        lines = []
+        backlog = []
```

Comment:
why don't you make `@backlog` instead of `backlog` local variable?

Reply:
In my opinion, local variable is better than instance variable to keep the scope of the variable small.
What do you think about this?

Reply:
@ko1 
In my opinion, local variable is better than instance variable to keep the scope of the variable small.
What do you think about this?

Reply:
In this case, `@last_backlog` and `@internal_info` are already used and I don't have any good reason why only `@backlog` is passed as an argument.

Reply:
"All in arguments" is another reasonable option for me.

Reply:
It doesn't make sense to me. The visibility control is not related to frequency. 

* `backlog` is only visible where the value is passed. Well controlled.
* `@internal_info` and `@last_backlog` are visible from any test class methods.

Visibility of these values should be same. At least I feel it should be same because there is no reason to change the visibility.

Reply:
In general, well-controlled visibility is good idea. So passing `backlog` as an argument is nice.
However, `@internal_info` and `@last_backlog` are already exposed, and maybe nobody modify them from test methods (in other words, we can believe test writers). In this background, there is no reason to prohibit exposing `backlog` and it reduces the code size.

Reply:
I fully persuaded myself of your opinion. Thanks!